From f96f774a25a857bc74309c308e5dae4fc91dc861 Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Sun, 21 May 2023 07:45:38 +0000
Subject: [PATCH] dts: renesas: Add plicsw & plmt nodes for OpenSBI

We need the PLMT and PLICSW nodes for initializing time and ipi
devices in OpenSBI at boot-time.

Signed-off-by: Yu Chien Peter Lin <peterlin@andestech.com>
---
 arch/riscv/boot/dts/renesas/r9a07g043f.dtsi | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/riscv/boot/dts/renesas/r9a07g043f.dtsi b/arch/riscv/boot/dts/renesas/r9a07g043f.dtsi
index b0796015e36b..d23aee504bdf 100644
--- a/arch/riscv/boot/dts/renesas/r9a07g043f.dtsi
+++ b/arch/riscv/boot/dts/renesas/r9a07g043f.dtsi
@@ -59,6 +59,22 @@ plic: interrupt-controller@12c00000 {
 		interrupts-extended = <&cpu0_intc 11 &cpu0_intc 9>;
 	};
 
+	plicsw: interrupt-controller@13000000 {
+		compatible = "andestech,plicsw";
+		reg = <0x00000000 0x13000000 0x00000000 0x00400000>;
+		interrupts-extended = <&cpu0_intc 0x03>;
+		interrupt-controller;
+		#address-cells = <2>;
+		#interrupt-cells = <2>;
+		riscv,ndev = <1>;
+	};
+
+	plmt: plmt0@110c0000 {
+		compatible = "andestech,plmt0";
+		reg = <0x00 0x110c0000 0x00 0x100000>;
+		interrupts-extended = <&cpu0_intc 0x07>;
+	};
+
 	l2cache: cache-controller@13400000 {
 		compatible = "andestech,ax45mp-cache", "cache";
 		reg = <0x0 0x13400000 0x0 0x100000>;
-- 
2.34.1

